lib/browser/desktop-capturer.js

```diff
@@ -10,27 +10,33 @@ const deepEqual = (a, b) => JSON.stringify(a) === JSON.stringify(b)
 // A queue for holding all requests from renderer process.
 let requestsQueue = []
 
-const electronSources = 'ELECTRON_BROWSER_DESKTOP_CAPTURER_GET_SOURCES'
-const capturerResult = (id) => `ELECTRON_RENDERER_DESKTOP_CAPTURER_RESULT_${id}`
+function defer () {
+  const ret = {}
+  ret.promise = new Promise((resolve, reject) => {
+    ret.resolve = resolve
+    ret.reject = reject
+  })
+  return ret
+}
 
-ipcMainInternal.on(electronSources, (event, captureWindow, captureScreen, thumbnailSize, fetchWindowIcons, id) => {
+ipcMainUtils.handle('ELECTRON_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', (event, captureWindow, captureScreen, thumbnailSize, fetchWindowIcons) => {
   const customEvent = eventBinding.createWithSender(event.sender)
   event.sender.emit('desktop-capturer-get-sources', customEvent)
 
   if (customEvent.defaultPrevented) {
-    event._replyInternal(capturerResult(id), [])
-    return
+    return []
   }
 
+  const deferred = defer()
```

Comment:
What if we did

```js
return new Promise((resolve, reject) => {
  ...
  const request = {
    ...
    resolve,
    reject,
    ....
  }
  ...
})
```

so we could remove this deferred pattern 🤔

Reply:
done